src/lib/userStorage/UserProfileStorage.js

```diff
@@ -0,0 +1,456 @@
+
+  walletAddressIndex = {}
+
+  //unecrypted profile field values
+  profile: { [key: string]: ProfileField } = {}
+
+  constructor(wallet: GoodWallet, profiledb: ProfileDB) {
+    // const seed = Uint8Array.from(Buffer.from(pkeySeed, 'hex'))
+    // this.privateKey = TextileCrypto.PrivateKey.fromRawEd25519Seed(seed)
+    this.wallet = wallet
+    this.profiledb = profiledb
+  }
+
+  /**
+   * read raw profile from database
+   */
+  async init() {
+    const rawProfile = await this.profiledb.getProfile()
+    this._setLocalProfile(await this._decryptProfileFields(rawProfile))
+  }
+
+  /**
+   * helper for setting value of local profile state
+   * @param newValue
+   * @private
+   */
+  _setLocalProfile(newValue) {
+    // this.subscribeProfileUpdates(newValue)
+    this.profile = newValue
+  }
+
+  /**
+   * helper for decrypt profile values
+   * @param profile
+   * @returns {Promise<{}>}
+   * @private
+   */
+  async _decryptProfileFields(profile): Promise<any> {
+    const outputProfile = {}
+    await Promise.all(
+      Object.keys(profile).map(
+        async item =>
+          typeof profile[item]?.value === 'string' &&
+          (outputProfile[item] = {
+            ...profile[item],
+            value: await this.profiledb.decryptField(profile[item]?.value),
+          }),
+      ),
+    )
+    return outputProfile
+  }
+
+  /**
+   * encrypt decrypted profile
+   * @param {*} profile
+   */
+  async _encryptProfileFields(profile): Promise<any> {
+    let encryptProfile = {}
+    await Promise.all(
+      Object.keys(profile).map(
+        async field =>
+          typeof profile[field]?.value === 'string' &&
+          (encryptProfile[field] = {
+            ...profile[field],
+            value: await this.profiledb.encryptField(profile[field]?.value),
+          }),
+      ),
+    )
+    return encryptProfile
+  }
+
+  /**
+   * saves a complete profile to the underlying storage
+   * @param {*} profile
+   * @param update
+   */
+  async setProfile(profile, update: boolean = false): Promise<any> {
+    if (update) {
+      const { getErrors, isValid, validate, ...profileFields } = profile
+      await Promise.all(
+        // eslint-disable-next-line no-return-await
+        Object.keys(profileFields).map(async key => await this.setProfileField(key, profileFields[key])),
+      )
+    } else {
+      const encryptedProfile = await this._encryptProfileFields(profile)
+      await this.profiledb.setProfile(encryptedProfile)
+    }
+  }
+
+  /**
+   * gets the current unencrypted field values in-memory
+   * @returns
+   */
+  getProfile(): Profile {
+    return Object.keys(this.profile).reduce((acc, currKey) => ({ ...acc, [currKey]: this.profile[currKey].value }), {})
+  }
+
+  /**
+   * updates profile fields in storage
+   * @param {*} fields
+   * @returns
+   */
+  async setProfileFields(fields: { [key: string]: ProfileField }): Promise<any> {
+    const encryptedFields = await this._encryptProfileFields(fields)
```

Comment:
same here you , you probably want to use a helper to modify each field 'display'

Reply:
fields encryption should be part of the profile storage
the **all stuff about profile** should be placed here